src/coreclr/src/pal/src/misc/sysinfo.cpp

```diff
@@ -29,7 +29,7 @@ Revision History:
 #include <inttypes.h>
 #include <sys/types.h>
 
-#if HAVE_SYSCONF
+#if HAVE_SYSCONF && !HAVE_SYSCTLBYNAME
```

Comment:
This complex condition combined with chained ifdef makes it hard to understand.

Would it be better to have a separate block like this?

```
#if HAVE_SYSCTLBYNAME
#include <sys/sysctl.h>
#endif
```
We may include `#include <sys/sysctl.h>` twice in some cases, but I think that's ok.

Reply:
+1, separated the condition.